### PR TITLE
feat: historacle migrations for v3.3.0 -> v3.4.0

### PR DESCRIPTION
**Pull request by adamewozniak**
## Description

Closes:  #1668

This is the migration for v3.3.0 -> v3.4.0 for historacle. 
Would love a tACK for this ❤️ 

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by adamewozniak** [comment]
Forgot to actually register the migration lol
I’ll add it in a minute

File: x/oracle/keeper/migrations.go
Comment (rbajollari) [inline x/oracle/keeper/migrations.go:29]: We should probably update these default values (I set them a while ago before a lot of discussions around what they should be around). Like do we want to start with the `HistoricStampPeriod` being equal to the `VotePeriod` for example? Right now the `DefaultHistoricStampPeriod` is a lot bigger than that.

Comment (adamewozniak) [reply x/oracle/keeper/migrations.go:29]: Yeah let's decide that in this PR - I think that might be reasonable. 24 hour historic stamps, 3 hour median?
Might want to talk about this in slack @rbajollari

File: app/upgrades.go
Comment (robert-zaremba) [inline app/upgrades.go:38]: This will be a direct upgrade, so let's simplify the name as we did with v3.2 and v3.1

Comment (robert-zaremba) [reply app/upgrades.go:38]:
```suggestion
	const planName = "v3.4"
```

File: x/oracle/keeper/migrations.go
Comment (robert-zaremba) [inline x/oracle/keeper/migrations.go:28]: Parameter value change is not a consensus version change.  Previously we needed to upgrade the consensus version, because it was breaking.
Let's rename this function to `HistoroacleParamsV3_4` (or something like that).

File: x/oracle/module.go
Comment (robert-zaremba) [inline x/oracle/module.go:161]:
```suggestion
```

Comment (robert-zaremba) [reply x/oracle/module.go:161]: Not needed, we don't bump module version.

File: x/oracle/module.go
Comment (robert-zaremba) [inline x/oracle/module.go:154]:
```suggestion
```

Comment (robert-zaremba) [reply x/oracle/module.go:154]: This was module version update, not app version update.

**Review by robert-zaremba — COMMENTED** [review]
1. The migration logic is a bit wrong: we don't do module upgrade, only the params change, hence the code in oracle/module.go won't be executed and should be removed (otherwise it's only a noise and will conflict with future module upgrade).
2. I'm changing the title to DNM, unless necessary, let's merge it after the 3.3 release
3. Please ping me for second review once the updates will be applied.

File: x/oracle/keeper/migrations.go
Comment (adamewozniak) [reply x/oracle/keeper/migrations.go:29]: Added in 4c99f1cc3c5ecd6cce3a2ef2eb27b28aacbf2b9c

File: x/oracle/keeper/migrations.go
Comment (adamewozniak) [reply x/oracle/keeper/migrations.go]: Won't this be consensus changing since we're changing how ABCI works? It definitely changes consensus for x/leverage once moser's PR gets merged

Comment (adamewozniak) [reply x/oracle/keeper/migrations.go]: In any case, I've changed it to just directly update those params: 
https://github.com/umee-network/umee/pull/1656/commits/1bebad098b87b03786c2f7d569c73651cd446d53

File: x/oracle/keeper/migrations.go
Comment (robert-zaremba) [inline x/oracle/keeper/migrations.go:28]: This function name is too general, while targets, 3.4 upgrade. Let's rename

Comment (robert-zaremba) [reply x/oracle/keeper/migrations.go:28]:
```suggestion
func (m Migrator) HistoracleParams3x4(ctx sdk.Context) error {
```

**Review by robert-zaremba — APPROVED** [review]
thanks for update, pre-approving, but let's rename the function, and keep DNM until we finalize 3.3

File: x/oracle/keeper/migrations.go
Comment (robert-zaremba) [reply x/oracle/keeper/migrations.go:29]: UP, let's address: https://github.com/umee-network/umee/pull/1656#pullrequestreview-1218175261

File: x/oracle/keeper/migrations.go
Comment (robert-zaremba) [inline x/oracle/keeper/migrations.go:28]: UP, let's address: https://github.com/umee-network/umee/pull/1656#pullrequestreview-1218175261

**Comment by adamewozniak** [comment]
I think there's a branch for v3.3 we can work out of if anything goes awry, so I believe this is g2g @robert-zaremba

File: app/upgrades.go
Comment (robert-zaremba) [inline app/upgrades.go:47]:
```suggestion
				ctx.Logger().Error("Unable to run v3.4 Migration!", "err", err)
```

**Review by robert-zaremba — APPROVED** [review]
utACK

**Pull request merged**
